RestApi/Persistence/DataBase/ApplicationDbContext.cs

```diff
@@ -4,14 +4,24 @@
 namespace RestApi.Persistence.DataBase;
 
 public class ApplicationDbContext : DbContext
+{
+    public DbSet<User> Users { get; set; }
+    public DbSet<ExpenseCategory> ExpenseCategory { get; set; }
+    public DbSet<Expense> Expenses { get; set; }
+    public DbSet<Income> Incomes { get; set; }
+
+    public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options) : base(options)
     {
-        public DbSet<User> Users { get; set; }    
-        public DbSet<Income> ExpenseCategory { get; set; }    
-        public DbSet<Expense> Expenses { get; set; }
-        public DbSet<Income> Incomes { get; set; }    
-       
-        public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options) : base(options)
-        {
-        }
-    
     }
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        base.OnModelCreating(modelBuilder);
+
+        modelBuilder.Entity<ExpenseCategory>().HasData(
+            new ExpenseCategory { Name = "Electronics" },
+            new ExpenseCategory { Name = "Transport" },
+            new ExpenseCategory { Name = "Products" }
```

Comment:
This is a failsafe for when the DB doesn't have default categories, right?

Reply:
They are default categories that already be on the DB

Reply:
nice